test-cypress/integration/explore/graphs.overview.spec.js

```diff
@@ -1,114 +1,131 @@
+        cy.get(`${EXPORT_GRAPHS_TABLE_ID} tbody tr`).should('have.length', count);
+    }
 
-        cy.get("div[paginations]").should('be.visible').should('contain', '3');
-        cy.verifyGraphExists("The default graph");
-    });
-
-    it('Test graphs overview pagination - switch pages', function () {
-        cy.wait(500);
-        cy.get("div[paginations] ul li a").eq(2).click().wait(500);
-        cy.get("li").should('contain', "2").should('be', 'active')
-        cy.verifyGraphExists("urn:11");
-        cy.get("div[paginations] ul li a").eq(3).click().wait(500);
-        cy.get("li").should('contain', "3").should('be', 'active')
-        cy.verifyGraphExists("urn:21");
-    });
+    //verifies that a certain graph exist within the current page in graphs overview.
+    function verifyGraph(graphName, exists = true) {
+        cy.get(EXPORT_GRAPHS_TABLE_ID).should(`${!exists ? 'not.' : ''}contain`, graphName);
+    }
 
+    function selectPage(number) {
+        return cy.get('div[paginations] ul li a').eq(number).click();
+    }
 
-    it('Test graphs overview pagination - test items per page', function () {
+    function selectItemFromMenu(number) {
         cy.get('[class="btn btn-secondary btn-sm dropdown-toggle"]').click();
-        cy.get('[class="dropdown-menu small"] li').eq(1).click();
-        cy.verifyGraphExists("urn:19");
-        cy.get('[class="btn btn-secondary btn-sm dropdown-toggle"]').click();
-        cy.get('[class="dropdown-menu small"] li').eq(2).click();
-        cy.verifyGraphExists("urn:1");
-        cy.verifyGraphExists("urn:21");
+        cy.get('[class="dropdown-menu small"] li').eq(number).click();
+    }
+
+    function confirmDelete() {
+        cy.get('.modal-footer .confirm-btn').click();
+        cy.get('.modal').should('not.be.visible');
+    }
+
+    context('Test graphs overview pagination', () => {
+        it('Should be visible', () => {
+            cy.get('div[paginations]')
+                .should('be.visible')
+                .and('contain', '3');
+            verifyGraph('The default graph');
+        });
+
+        it('Should switch pages', () => {
+            // Switch through pages and verify that the respective pager button is active.
+            selectPage(2).should('contain', '2')
+                .closest('li').should('be', 'active');
+            verifyGraph('urn:11');
+
+            selectPage(3).should('contain', '3')
+                .closest('li').should('be', 'active');
+            verifyGraph('urn:21');
+        });
     });
 
-    it('Test graphs overview search', function () {
-        cy.wait(1000);
-        cy.get('.form-control').type(graphToSearch);
-        cy.verifyGraphExists(graphToSearch);
-        cy.get("tbody").should('not.contain', "urn:11");
+    it('Test items per page switch menu', () => {
+        // Change number of items per page from the menu and verify that proper rows are present in
+        // the table.
+        selectItemFromMenu(1);
```

Comment:
It's not clear what 1 and 2 are, a comment can shed some light when reading the test. OR add properly named constants